File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"
#include <cmath>
namespace darts {

// TODO: add your solution here
    constexpr double OUTER = 10.0;
    constexpr double MIDDLE = 5.0;
    constexpr double INNER = 1.0;
    int score(double x, double y) {
        const auto r = std::sqrt(x * x + y * y);
        if (r > OUTER)
            return 0;
        else if (r > MIDDLE)
            return 1;
        else if (r > INNER)
            return 5;
        else
            return 10;
    }

}  // namespace darts

File: solutions/python/darts/1/darts.py
import math

def score(x: float, y:float) -> int:
    radius: float = math.sqrt(x**2 + y**2)
    match radius:
        case r if 5 < r <= 10: return 1
        case r if 1 < r <= 5: return 5
        case r if r <= 1: return 10
        case _: return 0

